### PR TITLE
Topic and City Models, Views, Urls, and Serializers Done

### PR DESCRIPTION
**Pull request by jennadean9**
all under cotrip_app

### CODE REVIEW

File: api/cotrip_app/views.py [hunk 1]
```diff
@@ -1,3 +1,29 @@
-from django.shortcuts import render
+from rest_framework import generics, permissions
+from .serializers import CitySerializer, TopicSerializer
+from .models import City, Topic
+from django.views.generic import ListView, DetailView
+from django.views.generic.edit import CreateView, DeleteView, UpdateView
 
-# Create your views here.
+
+class CityList(generics.ListCreateAPIView):
```
[the remark this answers is not in the document]
Comment (ZakkMan) [reply api/cotrip_app/views.py]: Maybe we don't have to worry about this just yet

**Comment by ZakkMan** [comment]
@jennadean9 can you also update your branch with `develop`?

File: api/api/urls.py [hunk 2]
```diff
@@ -14,14 +14,13 @@
...
-from django.urls import path, include
-from django.conf import settings
-from django.conf.urls.static import static
+from django.urls import path
+from django.conf.urls import include
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    # path('', include('cotrip_app.urls')),
-    path('', include('accounts.urls'))
```
Comment (ZakkMan) [inline api/api/urls.py]: Can you delete the commented out code? Should you be deleting the accounts urls here?

File: api/api/urls.py [hunk 3]
```diff
@@ -14,14 +14,13 @@
...
+    path('', include('cotrip_app.urls')),
+
 ]
 
 if settings.DEBUG:
```
Comment (ZakkMan) [inline api/api/urls.py]: This throws the following error:

```
NameError: name 'settings' is not defined
```

File: api/cotrip_app/urls.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django.urls import path
+from . import views
+
+urlpatterns = [
+    path('city/', views.CityList.as_view(), name='city_list'),
+    path('city/<int:pk>', views.CityDetail.as_view(), name='city_detail'),
+    path('city/new', views.CityList.as_view(), name='city_create'),
+    path('city/<int:pk>/edit', views.CityDetail.as_view(), name='city_edit'),
+    path('city/<int:pk>/delete',
```
Comment (ZakkMan) [inline api/cotrip_app/urls.py]: I don't think we want people to be able to create, update or delete these through the API

File: api/cotrip_app/urls.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+         views.CityDetail.as_view(), name='city_delete'),
+    path('topic/', views.TopicList.as_view(), name='topic_list'),
+    path('topic/<int:pk>', views.TopicDetail.as_view(),
+         name='topic_detail'),
+    path('topic/new', views.TopicList.as_view(), name='topic_create'),
+    path('topic/<int:pk>/edit',
+         views.TopicDetail.as_view(), name='topic_edit'),
+    path('topic/<int:pk>/delete',
+         views.TopicDetail.as_view(), name='topic_delete')
```
Comment (ZakkMan) [inline api/cotrip_app/urls.py]: Same here

File: api/cotrip_app/views.py [hunk 6]
```diff
@@ -1,3 +1,29 @@
...
+    queryset = City.objects.all()
+    serializer_class = CitySerializer
+    # permission_classes = (permissions.IsAuthenticated)
+
+
+class CityDetail(generics.RetrieveUpdateDestroyAPIView):
+    queryset = City.objects.all()
+    serializer_class = CitySerializer
+    # permission_classes = (permissions.IsAuthenticated)
```
Comment (ZakkMan) [inline api/cotrip_app/views.py]: I don't think we want people to be able to delete or update these through the API. Same with Create. We probably want to limit the functionality to list and create.

**Comment by colleenobrien3** [comment]
@Underwaterr i'll close it

**Pull request closed without merging**